### PR TITLE
[New] Set spatial reference.

### PR DESCRIPTION
**Pull request by fcamas**
## Description

This PR implements `Set spatial reference` in `Maps` category.
URL to  [README](https://devtopia.esri.com/runtime/common-samples/tree/main/designs/SetSpatialReference)

## Linked Issue(s)

- `swift/issues/`

## How To Test

- Pan and zoom around the map. 
- Observe how the map is displayed. 

## Screenshots

set-spatial-reference

### CODE REVIEW

File: Shared/Samples/Set spatial reference/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Choosing the correct spatial reference is important for ensuring accurate projection of data points to a map.  
+
+## How to use the sample
+
+Pan and zoom around the map. Observe how the map is displayed using the World Bonne spatial reference.
+
+## How it works
+
+1. Instantiate an `ArcGISMap` object using a spatial reference e.g. `ArcGISMap(SpatialReference.create(54024))`.
```
Comment (pgruenler) [inline Shared/Samples/Set spatial reference/README.md:17]:
```suggestion
1. Instantiate a `Map` object using a spatial reference e.g. `Map(spatialReference: SpatialReference(wkid: WKID(54024))`.
```

File: Shared/Samples/Set spatial reference/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+2. Instantiate a `Basemap` object using an `ArcGISMapImageLayer` object.
+3. Set the base map to the map.
+4. Set the map to a `MapView` object.
+
+The ArcGIS map image layer will now use the spatial reference set to the ArcGIS map (World Bonne (WKID: 54024)) and not it's default spatial reference.
```
Comment (pgruenler) [inline Shared/Samples/Set spatial reference/README.md:22]:
```suggestion
The ArcGIS map image layer will now use the spatial reference set to the map (World Bonne (WKID: 54024)) and not its default spatial reference.
```

File: Shared/Samples/Set spatial reference/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Relevant API
+
+* ArcGISMap
```
Comment (pgruenler) [inline Shared/Samples/Set spatial reference/README.md:26]:
```suggestion
* Map
```

File: Shared/Samples/Set spatial reference/README.metadata.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    "category": "Maps",
+    "description": "Specify a map's spatial reference.",
+    "ignore": false,
+    "images": [
+        "set-spatial-reference.png"
+    ],
+    "keywords": [
+        "WKID",
+        "project",
+        "ArcGISMap",
```
Comment (pgruenler) [inline Shared/Samples/Set spatial reference/README.metadata.json:11]:
```suggestion
        "Map",
```

File: Shared/Samples/Set spatial reference/README.metadata.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        "ArcGISMapImageLayer",
+        "Basemap",
+        "MapView",
+        "SpatialReference"
+    ],
+    "redirect_from": [],
+    "relevant_apis": [
+        "ArcGISMap",
```
Comment (pgruenler) [inline Shared/Samples/Set spatial reference/README.metadata.json:19]:
```suggestion
        "Map",
```

**Review by pgruenler — COMMENTED** [review]
Done with first review.

File: Shared/Samples/Set spatial reference/set-spatial-reference.png
Comment (pgruenler) [inline Shared/Samples/Set spatial reference/set-spatial-reference.png:1]: Please remove the "Licensed For Developer Use Only" watermark from the screen image.

Comment (fcamas) [reply Shared/Samples/Set spatial reference/set-spatial-reference.png:1]: Thank you for your suggestion; I’ve removed the watermark from the image.

**Review by pgruenler — DISMISSED** [review]

File: Shared/Samples/Set spatial reference/SetSpatialReferenceView.swift [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    }
+}
+
+private extension SpatialReference {
+    /// The spatial reference for the sample World Bonne (WKID: 54024).
+    static var worldBonne: Self { SpatialReference(wkid: WKID(54024)!)! }
+}
+
+private extension URL {
+    /// The URL to the World Cities image layer for the basemap.
```
Comment (CalebRas) [inline Shared/Samples/Set spatial reference/SetSpatialReferenceView.swift:39]:
```suggestion
    /// The URL to the World Cities map service for the image layer.
```

File: Shared/Samples/Set spatial reference/SetSpatialReferenceView.swift [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        return map
+    }()
+    
+    var body: some View {
+        MapView(map: map)
+    }
+}
+
+private extension SpatialReference {
+    /// The spatial reference for the sample World Bonne (WKID: 54024).
```
Comment (CalebRas) [inline Shared/Samples/Set spatial reference/SetSpatialReferenceView.swift:34]:
```suggestion
    /// The spatial reference for the sample, World Bonne (WKID: 54024).
```

Not sure about this one.... Dose this need a comma here?

**Review by CalebRas — COMMENTED** [review]
Looks good and works on all platforms. Great job on your first sample Fredy!

File: Shared/Samples/Set spatial reference/SetSpatialReferenceView.swift (hunk 7 above)
Comment (fcamas) [reply Shared/Samples/Set spatial reference/SetSpatialReferenceView.swift]: Yes, thank you for the feedback! I appreciate your input

**Review by CalebRas — APPROVED** [review]

**Review by pgruenler — APPROVED** [review]

**Pull request merged**